subprojects/dependency-lock/src/integTest/groovy/org/gradle/internal/dependencylock/generation/DependencyLockFileResolvedDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,437 @@
+
+        buildFile << mavenRepository(mavenRepo)
+        buildFile << customConfigurations(MYCONF_CUSTOM_CONFIGURATION)
+        buildFile << """
+            dependencies {
+                myConf 'foo:first:1.5'
+                myConf 'bar:first:2.+'
+            }
+        """
+        buildFile << copyLibsTask(MYCONF_CUSTOM_CONFIGURATION)
+
+        when:
+        succeedsWithEnabledDependencyLocking(COPY_LIBS_TASK_NAME)
+
+        then:
+        lockFile.text == """{
+${commentAndLockFileVersionJson()}
+  "projects": [
+    {
+      "path": ":",
+      "configurations": [
+        {
+          "name": "myConf",
+          "dependencies": [
+            {
+              "moduleId": "foo:first",
+              "requestedVersion": "1.5",
+              "lockedVersion": "1.5"
+            },
+            {
+              "moduleId": "foo:second",
+              "requestedVersion": "1.6.7",
+              "lockedVersion": "1.6.7"
+            },
+            {
+              "moduleId": "foo:third",
+              "requestedVersion": "1.5",
+              "lockedVersion": "1.5"
+            },
+            {
+              "moduleId": "bar:first",
+              "requestedVersion": "2.+",
+              "lockedVersion": "2.5"
+            },
+            {
+              "moduleId": "bar:second",
+              "requestedVersion": "2.6.7",
+              "lockedVersion": "2.6.7"
+            },
+            {
+              "moduleId": "bar:third",
+              "requestedVersion": "2.5",
+              "lockedVersion": "2.5"
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}"""
+        sha1File.text == 'c9dede601fb738575952da5dae120b2a35a9ff0d'
+    }
+
+    def "can create lock for conflict-resolved dependency"() {
+        given:
+        def fooSecondModule = mavenRepo.module('foo', 'second', '1.6.7').publish()
+        mavenRepo.module('foo', 'first', '1.5').dependsOn(fooSecondModule).publish()
+        mavenRepo.module('foo', 'second', '1.9').publish()
+
+        buildFile << mavenRepository(mavenRepo)
+        buildFile << customConfigurations(MYCONF_CUSTOM_CONFIGURATION)
+        buildFile << """
+            dependencies {
+                myConf 'foo:first:1.5'
+                myConf 'foo:second:1.9'
+            }
+        """
+        buildFile << copyLibsTask(MYCONF_CUSTOM_CONFIGURATION)
+
+        when:
+        succeedsWithEnabledDependencyLocking(COPY_LIBS_TASK_NAME)
+
+        then:
+        lockFile.text == """{
+${commentAndLockFileVersionJson()}
+  "projects": [
+    {
+      "path": ":",
+      "configurations": [
+        {
+          "name": "myConf",
+          "dependencies": [
+            {
+              "moduleId": "foo:first",
+              "requestedVersion": "1.5",
+              "lockedVersion": "1.5"
+            },
+            {
+              "moduleId": "foo:second",
+              "requestedVersion": "1.9",
+              "lockedVersion": "1.9"
+            }
```

Comment:
This is missing the edge from "foo:second:1.6.7 -> 1.9". So we have a requested component selector that is not represented in the data.
Imagine the case where instead the excluded version was "foo:second:1.+". There would be no way for us to use the lock file to satisfy the version range.

Reply:
done